src/org/isf/medicalstock/service/MedicalStockIoOperations.java

```diff
@@ -575,37 +572,37 @@ protected boolean updateMedicalWardQuantity(
 		ArrayList<Lot> lots = null;
 	
 		
-		List<Object[]> lotList = (List<Object[]>)lotRepository.findAllWhereMedical(medical.getCode());
-		lots = new ArrayList<Lot>();
-		for (Object[] object: lotList)
-		{
-			Lot lot = _convertObjectToLot(object);
-			
-			lots.add(lot);
-		}
-		
-		// remve empy lots
-		ArrayList<Lot> emptyLots = new ArrayList<Lot>();
-		for (Lot aLot : lots) {
-			if (aLot.getQuantity() == 0)
-				emptyLots.add(aLot);
-		}
-		lots.removeAll(emptyLots);
+		List<Object[]> lotList = lotRepository.findAllWhereMedical(medical.getCode());
```

Comment:
Isn't it possible to just return List<Lot> from repository? Than maybe you could avoid doing conversion?

Reply:
@marcoebbi that part should not be removed because it perform an important task on the lot (remove from the list the empty ones). So, until we introduce a different mechanism to do have the same result, we have to keep it. 

@all please pay attention to methods involving the Pharmacy (modules org.isf.medical* and related ones) and the Billing (org.isf.accounting.*)

Reply:
@mwithi In my opinion I did not change any behaviour here. Have a look at the repository method (findAllWhereMedical(medicalCode)). I just moved the empty lot filtering from code to database query. Should I still revert it?

Reply:
@marcoebbinghaus Ok, I've seen it, but two things: in some cases we could need to know all lots in the history, also the ones with zero quantity, so still we need to retrieve all of them from the DB; don't know if there is already another method for that, maybe we can parameter in case. Second, the query you wrote is slight different, because the original one calculates the LOT quantity as a SUM over a GROUP of movements concerning the same LOT.
Does it make sense also for you?

Reply:
@mwithi absolutely. Now that you mentioned it..I remember I struggled with that original query including the grouping and a corresponding replacement with JPQL. But I will have a second look. Thanks for the clarification.

Reply:
thanks @marcoebbinghaus, sure that mechanism should be improved in the future, other option is to leave it as is meanwhile.

Reply:
@mwithi as there is so much logic inside that native query I'm afraid I'm overwhelmed migrating it to use JPQL instead - so I would leave it as is (rollback my changes here). Maybe this can be separated within a following ticket.